app/models/post.rb

```diff
@@ -79,6 +79,7 @@ class Post < ApplicationRecord
   validates :media, polymorphism: { type: Media }, allow_blank: true
   # posting to a group, posting to a profile, and posting to an interest are mutually exclusive.
   validates_with ExclusivityValidator, over: %i[target_user target_group target_interest]
+  validates_with ExclusivityValidator, over: %i[uploads embed], if: :uploads || :embed
```

Comment:
I don't think you can do `if: :uploads || :embed`, the `||` will run when the class is defined and just return `:uploads` so it literally just becomes `if: :uploads`